### ISSUE TITLE
[ENHANCEMENT]  MathMatrix

### ISSUE THREAD

**Issue by ddoyen** [PremierLangage/components#46]
@nborie @esandier @mciissee 

Je lance cette issue pour discuter des améliorations à apporter au composant MathMatrix.

Alors voilà ce que j'ai noté pour l'instant (une partie des suggestions vient d'Etienne) :

* Actuellement, on ne peut ajouter/supprimer qu'une seule colonne et une seule ligne quand on redimensionne la matrice. Le nombre de colonnes et de lignes supplémentaires devrait correspondre à la taille du redimensionnement graphique.

* La zone de drag pour le redimensionnement n'est pas sur le crochet de la matrice, ce qui est un peu perturbant. On a envie de dragger le crochet.

* La zone de drag est aussi un peu étroite. Et le pointeur de la souris qui indique qu'on peut redimensionner pourait aussi être plus gros ou différent (une main ?).

* J'ai l'impression qu'on ne peut pas mettre que des valeurs numériques dans la matrice. Une lettre est transformée en 0. C'est bien ça ? Si oui, c'est un peu restrictif. Il faudrait au moins faire un mode texte/libre en plus du mode numérique (comme pour Input). Et pourquoi pas un mode MathInput light (sans le clavier) pour écrire des symboles mathématiques en TeX. 

* Il serait pas mal, lors des redimensionnements, de garder la mémoire des anciens coefficients. Histoire de ne pas perdre des coefficients qu'on vient d'entrer à cause d'une mauvaise manip de redimensionnement. Une façon de le faire, ce serait de considérer ce qui est affiché dans le composant comme une sous-matrice d'une plus grande matrice. Il y aurait cette grande matrice et une "vue" de cette grande matrice.

* Le style par défaut (fond gris, italique) n'est peut-être pas le meilleur choix. Bien sûr, cela se change facilement avec extracss, mais autant avoir un style par défaut qui soit d'usage courant. :-)

**Comment by mciissee** [comment]
Je suis d'accord pour ces modifications. Je ne me rappele pas si je limite les valeurs à des entiers mais je vais vérifier et pour un MathInput embarqué je vais essayer aussi si c'est possible
